### ISSUE TITLE
Coding error in GNTPP/models/prob_decoders/diffusion_modules /gaussian_diffusion.py

### ISSUE THREAD

**Issue by AnthonyChouGit** [BIRD-TAO/GNTPP#5]
There might be an error in `GaussianDiffusion.p_losses(self, x_start, t, noise=None)`, where the loss of the form $`\lVert \epsilon-\epsilon_\theta(\sqrt{\bar{\alpha}_t}x_0+\sqrt{1-\bar{\alpha}_t}\epsilon,t) \rVert^2 `$ is computed. The returned value of `self.denoise_fn(x_noisy, t, cond=cond)`, i.e. `x_recon`, is actually the predicted noise $` \epsilon_\theta(\sqrt{\bar{\alpha}_t}x_0+\sqrt{1-\bar{\alpha}_t}\epsilon,t) `$.

The code currently reads as:
`loss = torch.square(x_noisy - x_recon).sum()`

Since `x_noisy` is the corrupted sample, i.e. $`x_t`$, and I don't think $` x_t-\epsilon_\theta `$ makes any sense. So, in line 273, I think the correct code should be:
`loss = torch.square(noise- x_recon).sum()`

Please check it out.

**Comment by EDAPINENUT** [comment]
It is our mistakes to upload the code of false version. x noisy should be replaced by noise, in the correct version. Or the results of diffusion models cannot reach the state of the art.

**Comment by EDAPINENUT** [comment]
We will fix it asap

**Comment by EDAPINENUT** [comment]
Fixed
